src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -470,13 +497,33 @@
            netscape-comment-value)
           (utils/create-ca-extensions ca-name ca-serial ca-public-key))))
 
+(schema/defn read-infra-nodes
+    "Returns a list of infra node serials from the specified file organized as one item per line."
+    [cm-file :- schema/Str]
+    (line-seq (io/reader cm-file)))
+
+(schema/defn generate-compile-master-serials
+  "Given a list of infra nodes it will create a file containing 
+   serial numbers of their certificates (listed on separate lines).
+   It is expected have at least one entry (MoM)"
+   [{:keys [infra-nodes infra-node-serials-path signeddir]} :- CaSettings]
+
+   (with-open [wtr (io/writer infra-node-serials-path)]
+     (doseq [cm-node infra-nodes]
+       (try
+         (let [cm-serial  (-> (path-to-cert signeddir cm-node)
+                           (utils/pem->cert)
+                           (utils/get-serial))]
+            (.write wtr (str cm-serial))
+            (.newLine wtr))
+         (catch java.io.FileNotFoundException ex 
+            (log/warn (i18n/trs (str "Failed to find/load certificate for compile master: " cm-node)))))
+    )))
+
 (schema/defn generate-ssl-files!
   "Given the CA settings, generate and write to disk all of the necessary
   SSL files for the CA. Any existing files will be replaced."
   [ca-settings :- CaSettings]
-  (log/debug (str (i18n/trs "Initializing SSL for the CA; settings:")
-                  "\n"
-                  (ks/pprint-to-string ca-settings)))
```

Comment:
Is there a reason this log message was removed? (Sorry if I missed a comment about it, lots of review on this)